### DISCUSSION TITLE
Set property on all existing elements in a listing

### DISCUSSION THREAD

**Discussion by cloudflight-cweiss** [apple/pkl#530] [Q&A] [answered]
Maybe it is already mentionend in the language reference somewhere but I could not find it so here it goes...

I have a setup where I preset a number of "default tenants" which are used on mostly all deployed instances.
Basic template is this:
```pkl
module listmodify

tenants: Listing<Tenant>

class Tenant {
  tenant: String
  login: Login
}

class Login {
  username: String
  password: String
}
```

Now i have a file for defining the default tenants:

```pkl
amends "listmodify-template.pkl"

tenants {
  new {
    tenant = "AT"
  }
  new {
    tenant = "DE"
  }
  new {
    tenant = "CZ"
  }
}
```

And now i want to generate a fully filled out config for my local environment for example:
```pkl
amends "default-tenants.pkl"

local defaultLogin = new Login {
  username = "admin"
  password = "admin"
}

// TODO: Set login property on all existing tenant elements
tenants {

}
```

How can I set the login property on all existing elements in the "tenants" listing?

**Comment by bioball** [comment, marked as answer]
Use a [member predicate](https://pkl-lang.org/main/current/language-reference/index.html#member-predicates), where the predicate is simply `true`:

```pkl
amends "default-tenants.pkl"

local defaultLogin = new Login {
  username = "admin"
  password = "admin"
}

tenants {
  [[true]] {
    login = defaultLogin
  }
}
```

**Reply by cloudflight-cweiss** [reply]
Ah thanks.....I was trying to use member predicates as the solution but did not think of just using true as predicate to select all members :)
